core/templates/dev/head/pages/story_editor/main_editor/StoryEditorDirective.js

```diff
@@ -40,6 +43,32 @@ oppia.directive('storyEditor', [
             $scope.editableDescriptionIsEmpty = (
               $scope.editableDescription === '');
             $scope.storyDescriptionChanged = false;
+            if ($scope.storyContents) {
+              $scope.nodes = $scope.storyContents.getNodes();
+              if (!$scope.idOfNodeToEdit) {
```

Comment:
In what case would this check not trigger? (Can _init happen more than once?)

Reply:
Yeah, initially, _init() is called with interstitial values, at which point storyContents is null.

Reply:
That addresses the check on line 46, but I was talking about the check on line 48.

Reply:
Oh, this was just to display the initial node when loading the story page for the first time. If this was not there, every time an edit was done to any nodes, it reset to viewing the edit page for the initial node.

Reply:
I see. But why is init() called after every apply change? That seems a bit weird to me.

Reply:
You might want to somehow distinguish the concepts of "initial page initialization" and "update the frontend after each change" -- init() typically refers to the former, unless you're talking about some kind of subcomponent that is actually removed and redrawn.